jasmine-standalone-2.3.4 (4)/spec/GameSpec.js

```diff
@@ -0,0 +1,243 @@
+describe("Game", function() {
+  var game;
+  var player;
+
+  beforeEach(function() {
+    game = new Game();
+    player = new Player();
+    player.newGame(game);
+  });
+
+  describe("beginning of the game", function() {
+
+    it("has empty array of frames", function() {
+      expect(game.frames).toEqual([]);
+    });
+
+    it("has empty array of pairs", function() {
+      expect(game.pairs).toEqual([]);
+    });
+  });
+
+  describe("many-rolls basic game", function() {
+
+    beforeEach(function() {
+      player.roll(5);
+      player.roll(6);
+      player.roll(7);
+      player.roll(8);
+    });
+
+    it("adds frames to frames", function() {
+      expect(game.frames).toEqual([1, 1, 2, 2]);
+    });
+
+    it("adds objects to pairs", function() {
+      expect(game.pairs).toEqual([{1:5}, {1:6}, {2:7}, {2:8}])
+    });
+
+    it("returns last frame", function() {
+      expect(game._lastFrame()).toEqual(2);
+    });
+  });
+
+  describe("many-rolls game - counting score", function() {
+
+    beforeEach(function() {
+      player.roll(4);
+      player.roll(4); //
+      player.roll(5);
+      player.roll(3); //
+      player.roll(4);
+      player.roll(5); //
```

Comment:
Do we need to do all of these rolls? What does rolling 10 times test that rolling twice does not? 

This is another sign that we should be refactoring. Our tests should be focussed on specific, isolated actions - I roll and it records the score. I roll twice and it records a frame score. I score two frames and the game adds them together correctly. I roll a strike and it records the bonuses correctly.

If we're having to repeat a lot of actions in test setup then we're either introducing a lot of points of failure (brittle tests) or we are not really adding anything with many of the actions (redundant tests).